core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -163,7 +163,10 @@ protected boolean isValidIdentifier(TableIdentifier identifier) {
 
   @Override
   protected TableOperations newTableOps(TableIdentifier identifier) {
-    return new HadoopTableOperations(new Path(defaultWarehouseLocation(identifier)), conf);
+    return new HadoopTableOperations(
+        new Path(defaultWarehouseLocation(identifier)),
+        conf.get(ConfigProperties.CUSTOM_FILE_IO_IMPL),
```

Comment:
I think that the IO implementation should be passed in instead of the class name. `HadoopFileIO` only needs to be instantiated once per catalog and can be shared. Passing the instance from a `Catalog` into `TableOperations` allows the catalog implementation to choose whether to instantiate `FileIO` per table or not.